### PR TITLE
Refactor tests into structured test suites

### PR DESCRIPTION
**Pull request by MohamedSabthar**
## Purpose

Related to: https://github.com/ballerina-platform/ballerina-library/issues/7030
Related PR: https://github.com/ballerina-platform/module-ballerina-graphql/pull/2049

## Checklist
- [x] Linked to an issue
- [x] Added tests
- [x] Checked native-image compatibility
- [x] No `commons` package changes (if there are any, please update the GraphQL version in [GraphQL tools](https://github.com/ballerina-platform/graphql-tools) and [Ballerina dev tools](https://github.com/ballerina-platform/ballerina-dev-tools))
- [x] No `compiler` package changes (if there are any, please update the GraphQL version in [Ballerina dev tools](https://github.com/ballerina-platform/ballerina-dev-tools))

### CODE REVIEW

**Review by NipunaRanasinghe — DISMISSED** [review]

File: ballerina-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+- **graphql-client-test-suite**: Covers test cases for the Ballerina GraphQL client.
+- **graphql-dataloader-test-suite**: Covers test cases related to the DataLoader implementation.
+- **graphql-interceptor-test-suite**: Covers test cases for interceptors.
+- **graphql-security-test-suite**: Covers test cases for security.
+- **graphql-subgraph-test-suite**: Covers federation and subgraph-related implementation.
+- **graphql-subscription-test-suite**: Covers subscription-related functionality.
+- **graphql-service-test-suite**: Covers the remaining overall GraphQL specification.
+
+### Test Structure
+
```
Comment (DimuthuMadushan) [inline ballerina-tests/README.md:15]:
```suggestion
```

File: ballerina-tests/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+- **Listeners**: Test listeners are defined in the `<suite>/tests/listeners.bal` file.
+- **Services**: Test services are defined in the `<suite>/tests/services.bal` file.
+- **Records, Object Types, and Values**: Ballerina records, object types, and values are defined in the corresponding Ballerina files:
+  - `<suite>/tests/records.bal`
+  - `<suite>/tests/object_types.bal`
+  - `<suite>/tests/values.bal`
+
+### Handling Large Documents and JSON Files
```
Comment (DimuthuMadushan) [inline ballerina-tests/README.md:25]: Shall we check whether we need to use sentence-case here?

Comment (MohamedSabthar) [reply ballerina-tests/README.md:25]: Previous README.md file created for the the tests had Title case, hence following the same here.

**Review by DimuthuMadushan — APPROVED** [review]

File: ballerina-tests/README.md (hunk 1 above)
Comment (MohamedSabthar) [reply ballerina-tests/README.md:15]: Fixed with: [aeb8e8e](https://github.com/ballerina-platform/module-ballerina-graphql/pull/2050/commits/aeb8e8ecb4a98c28441beaee1416b27c5f9fe2a6)

**Pull request merged**